test/e2e/e2e.go

```diff
@@ -40,3 +45,73 @@ func SetupWithNamespace(t *testing.T, namespace string) *test.Clients {
 	}
 	return clients
 }
+
+// CreateNamespace creates a namespace to run the integration tests.
+func CreateNamespace(t *testing.T, clients *test.Clients, namespace string) {
+	_, err := clients.KubeClient.Kube.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+	if err != nil && apierrs.IsNotFound(err) {
+		// Create the namespace if not available
+		if _, err = clients.KubeClient.Kube.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta:
+			metav1.ObjectMeta{Name: namespace}}); err != nil {
+			t.Fatalf("Failed to create the namespace %s: %v", namespace, err)
+		}
+	}
+}
+
+// TestKnativeServingCreation creates a KnativeServing to see if it can reach the READY status.
+func TestKnativeServingCreation(t *testing.T, clients *test.Clients, names test.ResourceNames) {
+	if _, err := resources.CreateKnativeServingReady(t, clients.KnativeServingAlphaClient, names,
+		resources.IsKnativeServingReady); err != nil {
+		t.Fatalf("KnativeService %q failed to get to the READY status: %v", names.KnativeServing, err)
+	}
+}
+
+// TestDeploymentRecreation verify whether all the deployments for knative serving are able to recreate, when they are deleted.
+func TestDeploymentRecreation(t *testing.T, clients *test.Clients, names test.ResourceNames) *v1.DeploymentList {
+	dpListSave, err := clients.KubeClient.Kube.AppsV1().Deployments(names.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get any deployment under the namespace %q: %v",
+			test.ServingOperatorNamespace, err)
+	}
+	if len(dpListSave.Items) == 0 {
+		t.Fatalf("No deployment under the namespace %q was found",
+			test.ServingOperatorNamespace)
+	}
+	// Delete the deployments one by one to see if they will be recreated.
+	for _, deployment := range dpListSave.Items {
+		if err := clients.KubeClient.Kube.AppsV1().Deployments(deployment.Namespace).Delete(deployment.Name,
+			&metav1.DeleteOptions{}); err != nil {
+			t.Fatalf("Failed to delete deployment %s/%s: %v", deployment.Namespace, deployment.Name, err)
+		}
+		if _, err = resources.WaitForDeployment(clients, deployment.Name, deployment.Namespace,
+			resources.IsDeploymentAvailable, "DeploymentIsAvailable"); err != nil {
+			t.Fatalf("The deployment %s/%s failed to reach the desired state: %v",
+				deployment.Namespace, deployment.Name, err)
+		}
+		if _, err := resources.WaitForKnativeServingState(clients.KnativeServingAlphaClient, test.ServingOperatorName,
+			resources.IsKnativeServingReady); err != nil {
+			t.Fatalf("KnativeService %q failed to reach the desired state: %v", test.ServingOperatorName, err)
+		}
+		t.Logf("The deployment %s/%s reached the desired state.", deployment.Namespace, deployment.Name)
+	}
+	return dpListSave
```

Comment:
I wouldn't return this. Let the tests be independent. Don't pass a return from one to another.

Reply:
since we do not need to delete the deployments in the teardown, we do not need the return any mor.